src/NJsonApiCore/Infrastructure/Delta.cs

```diff
@@ -129,26 +126,42 @@ public T ToObject()
             return t;
         }
 
-        private Dictionary<string, Action<T, object>> ScanForProperties()
+        private Dictionary<string, Action<object, object>> ScanForProperties()
         {
-            return typeof(T)
-                .GetProperties()
-                .Where(pi => ObjectPropertyValues.ContainsKey(pi.Name))
-                .ToDictionary(pi => pi.Name, pi => pi.ToCompiledSetterAction<T, object>());
+            // set1 contains simple properties setters that are not related resources
+            var set1 = ObjectPropertyValues
+                .Where(opv => _mapping.PropertySetters.ContainsKey(opv.Key))
+                .ToDictionary(opv => opv.Key, opv => _mapping.PropertySetters[opv.Key]);
+            // set2 contains simple properteis setters for 1:1 related resources
+            var set2 = ObjectPropertyValues
+                .Join(
+                    _mapping.Relationships.Where(r => !r.IsCollection),
+                    opv => opv.Key,
+                    r => r.RelationshipName,
+                    (opv, r) => new KeyValuePair<string, Action<object, object>>(opv.Key, (Action<object,object>)r.RelatedProperty.SetterDelegate))
+                .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+            return set1.Concat(set2).ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
```

Comment:
Ah, that is not the case for delta. Delta can contain only SingleResourceIdentifiers, so nesting information is not passed on Delta (that can contain only resource id, but not resource attribute values, hence no resources nested).